maze-consulting/SFCC_Klaviyo_Maze
# Feature/kscu 78 list subscriptions

## Description

This update handles the ability to add and configure Email & SMS opt-in check boxes within Checkout _(Ex: [KSCU-78](https://themazegroup.atlassian.net/browse/KSCU-78))_. Development for this ticket included the following: 

- Code snippets for a checkbox for users to opt into SMS communication
- Code snippets for a checkbox for users to opt into Email communication
- Documentation to support the use of these options _([written & stored here](https://docs.google.com/document/d/1OUjUCAwov12HQHTpqeheTtgRS3Cd0P6kc9EaG9pwm7I/edit))_


## Assumptions & Acceptance Criteria Outlined in Ticket

The subscription event is submitted upon the continued checkout button. Additionally, users will have the ability to select either SMS or Email, both SMS and Email, or none. The client will be responsible for modifying displayed checkbox text as needed. 

A/C points: 

- As a user, I can opt into email & SMS data, and will receive communication from the vender.
- As an admin, I can see collected email & SMS dta being communicated to Klaviyo V3 APIs
- As an admin, I can configure list IDs for data being sent to Klaviyo



## Manual Testing Steps

Manual testing was conducted on this work, which included:

-  using checkbox snippets in the checkout process (ex: interacting with checking on and off & proceeding through checkout)
- setting breakpoints to evaluate collected data and responses within the code before and after API calls were sent
- cross checking the Klaviyo Dashboard for subscribed users (both before and after users accepted SMS terms & email notifications)
- testing subscription notifications with real phone numbers & emails as part of confirmation, etc. 
- Running ES lint commands to check virtual environments for & adjust any flagged linter errors 
- importing meta data (system objects & services) into a different sandbox and confirming functionality between 21.7 / 22.7 compatibility modes
